loopy/target/c/__init__.py

```diff
@@ -172,6 +172,45 @@ def _preamble_generator(preamble_info):
             yield ("04_%s" % func_name, func_body)
             yield undef_integer_types_macro
 
+    for func in preamble_info.seen_functions:
+        if func.name == "int_pow":
+            base_ctype = preamble_info.kernel.target.dtype_to_typename(
+                    func.arg_dtypes[0])
+            exp_ctype = preamble_info.kernel.target.dtype_to_typename(
+                    func.arg_dtypes[1])
+            res_ctype = preamble_info.kernel.target.dtype_to_typename(
+                    func.res_dtypes[0])
+
+            if func.arg_dtypes[1].numpy_dtype.kind == "u":
+                signed_exponent_preamble = ""
+            else:
+                signed_exponent_preamble = """
+              if (n < 0) {
+                x = 1.0/x;
+                n =  -n;
+              }"""
```

Comment:
Shouldn't this be guarded so that it only applies to FP arguments?

Reply:
For int-typed base and negative exponents, according to loopy-spec that should be undefined behavior and so the above code should be fine IMO. Let me know if you disagree.

Reply:
Yeah, I guess you're right. We don't know much about the positivity of `n`, so for negative `n` and integer types, I guess there's not much we can do. I'm mainly concerned about the `1.0` being used in an integer context, and how badly a compiler might complain about that.

Reply:
Sec 6.3.1.8 of https://web.archive.org/web/20181230041359if_/http://www.open-std.org/jtc1/sc22/wg14/www/abq/c17_updated_proposed_fdis.pdf:
> First, if the corresponding real type of either operand is long double, the other operand
is converted, without change of type domain, to a type whose corresponding real type is
long double.
Otherwise, if the corresponding real type of either operand is double, the other operand is
converted, without change of type domain, to a type whose corresponding real type is double.
Otherwise, if the corresponding real type of either operand is float, the other operand is
converted, without change of type domain, to a type whose corresponding real type is float.

So there would be conversions but no complaining, I guess

Reply:
Thanks! In that case, sure, let's try it and see what happens.